### ISSUE TITLE
coldcard cannot send when p2wpkh-p2sh change output is present

### ISSUE THREAD

**Issue by SomberNight** [spesmilo/electrum#5612]
Tested on latest release (3.3.8) and on git HEAD (https://github.com/spesmilo/electrum/commit/9372eacc291ab3fb35c762eba0341d95571863ff).

Using a coldcard standard wallet, `p2wpkh-p2sh` script type, I am creating a transaction with
1 input and 2 outputs. Only one of the outputs is `is_mine` (and it is more specifically `is_change`).

coldcard firmware version: 2.1.2
ckcc-protocol python package: 0.7.9

I'm getting "Change Fraud" on the device screen.
On host PC:
```
E | plugins.coldcard.coldcard.Coldcard_KeyStore | 
Traceback (most recent call last):
  File "...\electrum\electrum\plugins\coldcard\coldcard.py", line 541, in sign_transaction
    resp = client.sign_transaction_poll()
  File "...\electrum\electrum\plugins\coldcard\coldcard.py", line 245, in sign_transaction_poll
    return self.dev.send_recv(CCProtocolPacker.get_signed_txn(), timeout=None)
  File "C:\Users\User\AppData\Roaming\Python\Python37\site-packages\ckcc\client.py", line 159, in send_recv
    return CCProtocolUnpacker.decode(resp)
  File "C:\Users\User\AppData\Roaming\Python\Python37\site-packages\ckcc\protocol.py", line 182, in decode
    return d(msg)
  File "C:\Users\User\AppData\Roaming\Python\Python37\site-packages\ckcc\protocol.py", line 196, in err_
    raise CCProtoError("Remote Error: " + str(msg[4:], 'utf8', 'ignore'), msg[4:])
ckcc.protocol.CCProtoError: Remote Error: Output#0: Change output is fraudulent
I | plugins.coldcard.coldcard.Coldcard_KeyStore | Remote Error: Output#0: Change output is fraudulent
```

(it might actually be a bug in the coldcard firmware; unsure)

@peter-conalgo: I found this while testing https://github.com/spesmilo/electrum/pull/5440 but turns out it's unrelated to that PR. As that PR moves around a lot of code, feel free to fix it on top of it if you like -- please don't force-push that branch now though, I have a local copy with some changes.

**Comment by peter-conalgo** [comment]
Really looks like the Coldcard has an issue w/ this type of change output. Please send me an example PSBT with the the problem. Gist or here or email is fine. Thanks!

**Comment by SomberNight** [comment]
See e.g.:
```
cHNidP8BAHMCAAAAAWfDyBTzSTCNkH0sqWbIphDFWso8aiecCWq22szr6ot/AAAAAAD9////AnSFAQAAAAAAF6kUxHQz38Z+PglvF4PhxxBhQPAHJEKHoIYBAAAAAAAXqRR1GpUY5KbuZjHzBdYoHgmAjo80eoe+DxgAAAEBILoMAwAAAAAAF6kUR1zuipfw28EwcnrofJ19NXbTz/qHIgYDMapcB8zKMKhxyhbHMMIczkpE3DiH46bEluv6zCcFDmcYsnaNLzEAAIABAACAAAAAgAAAAAACAAAAAQQWABRKvD2jXJXkfGLQhgAJ3PhGyg5z/wAiAgOKVP/CztdS/1jY1xjSJlxN4PLu0+Pea2+N7KFjNL0xPhiydo0vMQAAgAEAAIAAAACAAQAAAAEAAAABABYAFANtyPuVUxKYA+7WX7j8n6MLA6ZXAAA=
```

Regardless of example, you should be able to reproduce from original description; or if it works for you then that too would be valuable information.

**Comment by SomberNight** [comment]
I think this must be a regression in the firmware; I must have tested this around https://github.com/spesmilo/electrum/pull/4784 and then it was working

**Comment by peter-conalgo** [comment]
Yes, it was a regression; sorry. It's now fixed in version 2.1.3 available at [coldcardwallet.com/docs/upgrade](https://coldcardwallet.com/docs/upgrade)

(Our test code also upgraded to exercise these change-output cases better!)

**Comment by SomberNight** [comment]
Thanks
